airflow-core/src/airflow/models/mappedoperator.py

```diff
@@ -173,3 +151,88 @@ def get_extra_links(self, ti: TaskInstance, name: str) -> str | None:
         if not link:
             return None
         return link.get_link(self, ti_key=ti.key)  # type: ignore[arg-type]
+
+
+@functools.singledispatch
+def get_mapped_ti_count(task: DAGNode, run_id: str, *, session: Session) -> int:
+    raise NotImplementedError(f"Not implemented for {type(task)}")
+
+
+# https://github.com/python/cpython/issues/86153
+# While we support Python 3.9 we can't rely on the type hint, we need to pass the type explicitly to
```

Comment:
We dropped support for Py 3.9 on main recently

Reply:
Let me try to change this in main too.